scratch/ScratchIslandApp/SampleApp/App.cpp

```diff
@@ -33,6 +26,44 @@ namespace winrt::SampleApp::implementation
         HighContrastAdjustment(::winrt::Windows::UI::Xaml::ApplicationHighContrastAdjustment::None);
     }
 
+    void App::Initialize()
+    {
+        const auto dispatcherQueue = winrt::Windows::System::DispatcherQueue::GetForCurrentThread();
+        if (!dispatcherQueue)
+        {
+            _windowsXamlManager = xaml::Hosting::WindowsXamlManager::InitializeForCurrentThread();
+        }
+        else
+        {
+            _isUwp = true;
+        }
+    }
+
+    void App::Close()

```

Comment:
behavior that was originally trapped in the toolkit application is herein merged (same for TerminalApp)